### ISSUE TITLE
`treeshake.moduleSideEffects` incorrectly eliminates `export * from "external"`

### ISSUE THREAD

**Issue by Swatinem** [rollup/rollup#2904]
found here: https://github.com/Swatinem/rollup-plugin-dts/pull/34

- Rollup Version: 1.14.3
- Operating System (or Browser): arch
- Node Version: 11.15

### How Do We Reproduce?

No repl, since you can’t set `treeshake.moduleSideEffects` options.

But its as simple as:
```ts
export * from "rollup";
```

with a rollup config:

```ts
treeshake: { moduleSideEffects: "no-external" },
external: ["rollup"],
```

### Expected Behavior

external `export *` should not be touched, no matter what

### Actual Behavior

the whole `export *` statement is removed.

**Comment by lukastaegert** [comment]
> luckily @lukastaegert is super responsive with all rollup issues :-) ❤️

Thanks for the kind words, let's see how response times change once my paternal leave ends 😉
Fix at #2905

**Comment by aleclarson** [comment]
@lukastaegert Keep up the great work! ❤️
